devito/function.py

```diff
@@ -710,12 +735,16 @@ def __init__(self, *args, **kwargs):
             d = Dimension(name='d')
             self.coordinates = Function(name='%s_coords' % self.name,
                                         dimensions=[self.indices[-1], d],
-                                        shape=(self.npoint, self.grid.dim))
+                                        shape=(self.npoint, self.grid.dim),
+                                        space_order=0)
```

Comment:
I wouldn't add that yet and leave it to default, I am still looking at extending FD to SparseFunctions. This seem like a hacky assumption for the compiler.

Reply:
I thought ``coordinates`` for a ``SparseFunction`` could only be a ``space_order=0`` field. There's no hidden hack in here, so I can drop it you prefer. I remember I added it because in the upcoming domain-allocation switch it was fixing something, but I don't remember what that was. So this is a cherry-picked commit that it felt appropriate to have it in master ASAP. 

Can we infer the coordinates space order from the ``SparseFunction`` initialisation ?

Reply:
My bad, added the comment at the wrong place, coordinates are indeed always zero order, the SparseFunction may not.

Reply:
👍